appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChartData.java

```diff
@@ -0,0 +1,137 @@
+    private static final String PROPERTY_COLOR = "Color";
+    private static final String PROPERTY_LABEL = "Label";
+    private static final String PROPERTY_PAIRS = "ElementsFromPairs";
+
+    // Represents the Chart data icon
+    private Image iconWidget;
+
+    protected MockChart chart;
+    protected MockChartDataModel chartDataModel;
+
+    private String currentElements = "";
+
+    /**
+     * Creates a new instance of a Mock Chart Data component.
+     *
+     * @param editor editor of source file the component belongs to
+     * @param type  type string of the component
+     * @param icon  icon of the component
+     */
+    MockChartData(SimpleEditor editor, String type, ImageResource icon) {
+        super(editor, type, icon);
+
+        iconWidget = new Image(icon);
+        iconWidget.setHeight("100");
+        iconWidget.setWidth("60");
+
+        initComponent(iconWidget);
+    }
+
+    /**
+     * Adds the Mock Chart Data component to the specified Mock Chart component
+     * @param chart  Chart Mock component to add the data to
+     */
+    public void addToChart(MockChart chart) {
+        // Hide widget (MockChartData component handled by Chart)
+        iconWidget.setVisible(false);
+        iconWidget.setHeight("0");
+        iconWidget.setWidth("0");
+
+        // Set references for Chart view and Chart model
+        this.chart = chart;
+        this.chartDataModel = chart.createDataModel();
+
+        // Set the properties to the Data Series
+        setDataSeriesProperties();
+
+        // Refresh the Chart view
+        refreshChart();
+    }
+
+    @Override
+    protected boolean isPropertyVisible(String propertyName) {
+        // Hide HEIGHT and WIDTH properties (not needed for Chart Data)
+        if (propertyName.equals(PROPERTY_NAME_HEIGHT) ||
+                propertyName.equals(PROPERTY_NAME_WIDTH)) {
+            return false;
+        }
+
+        return super.isPropertyVisible(propertyName);
+    }
+
+    @Override
+    protected void onSelectedChange(boolean selected) {
+        super.onSelectedChange(selected);
+        removeStyleDependentName("selected"); // Force remove styling
+    }
+
+    @Override
+    public void onRemoved() {
+        super.onRemoved();
+        chartDataModel.removeDataSeriesFromChart();
+        refreshChart();
+    }
+
+    /**
+     * Sets the current elements of the Chart Data component.
+     *
+     * @param text  new elements text
+     */
+    private void setElementsFromPairsProperty(String text){
+        currentElements = text;
+
+        chartDataModel.setElements(currentElements);
+    }
+
+    @Override
+    public void onPropertyChange(String propertyName, String newValue) {
+        super.onPropertyChange(propertyName, newValue);
+
+        // No Chart Model exists (Data not yet added to Chart), simply
+        // return from the method without processing property adding.
+        if (chartDataModel == null) {
```

Comment:
Is there a better way to handle this?

As soon as the Data component is initialized in GWT, it is not actually on the Chart. However, the properties rely on objects instantiated by a Chart component.